src/main/java/hw36/Cart.java

```diff
@@ -0,0 +1,27 @@
+package hw36;
+
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
```

Comment:
You should choose only 1 option how to configure this class as a Spring component - or using annotation (@Component) or as Java configuration - as you did `@Bean
    public Cart cart() { return new Cart(); }`